scripts/collection_from_items.py

```diff
@@ -112,28 +118,51 @@ def main() -> None:
 
     s3_client = client("s3")
 
-    files_to_read = list_json_in_uri(uri, s3_client)
+    files_to_read = list_files_in_uri(uri, (".json", "_footprint.geojson"), s3_client)
 
     start_time = time_in_ms()
+    polygons = []
     for key, result in get_object_parallel_multithreading(
         bucket_name_from_path(uri), files_to_read, s3_client, arguments.concurrency
     ):
-        item_stac = json.loads(result["Body"].read().decode("utf-8"))
-
-        if not arguments.collection_id == item_stac.get("collection"):
-            get_log().trace(
-                "skipping: item.collection != collection.id",
-                file=key,
-                action="collection_from_items",
-                reason="skip",
-            )
-            continue
-
-        collection.add_item(item_stac)
-        get_log().info("item added to collection", item=item_stac["id"], file=key)
+        content = json.loads(result["Body"].read().decode("utf-8"))
+
+        if key.endswith(".json"):
+            if not arguments.collection_id == content.get("collection"):
+                get_log().trace(
+                    "skipping: item.collection != collection.id",
+                    file=key,
+                    action="collection_from_items",
+                    reason="skip",
+                )
+                continue
+            collection.add_item(content)
+            get_log().info("item added to collection", item=content["id"], file=key)
+        elif key.endswith("_footprint.geojson"):
+            get_log().debug(f"adding geometry from {key}")
+            geom = shapely.geometry.shape(content["features"][0]["geometry"])
+            polygons.append(geom)
+
+    capture_area = geojson.Feature(geometry=shapely.ops.unary_union(polygons), properties={})
+    with TemporaryDirectory() as tmp_path:
+        tmp_capture_area_path = os.path.join(tmp_path, CAPTURE_AREA_FILE_NAME)
+        write(
+            tmp_capture_area_path,
+            json.dumps(capture_area).encode("utf-8"),
```

Comment:
Writing from a file descriptor should be more efficient than first reading it into memory. But I'm not sure if this sort of optimisation is worth it.